src/ucp/core/ucp_mm.c

```diff
@@ -1385,16 +1400,53 @@ ucs_status_t ucp_mem_reg_md_map_update(ucp_context_h context)
     context->alloc_md_index[alloc_mem_type] = alloc_memh->alloc_md_index;
     ucp_memh_put(context, alloc_memh);
 
-    status = ucp_memh_register(context, &reg_memh.memh,
-                               context->reg_md_map[alloc_mem_type], &buff,
-                               sizeof(buff), alloc_mem_type,
+    return UCS_OK;
+}
+
+static void ucp_mem_reg_md_map_exclude_cuda(ucp_context_h context,
+                                            ucs_memory_type_t mem_type,
+                                            ucp_md_map_t *md_map_included,
+                                            ucp_md_map_t *md_map_excluded)
+{
+    ucp_md_index_t md_index;
+
+    ucs_for_each_bit(md_index, context->reg_md_map[mem_type]) {
+        if (context->tl_mds[md_index].attr.reg_mem_types &
+            UCS_BIT(UCS_MEMORY_TYPE_CUDA)) {
```

Comment:
this can also be true for IB.
can we try some other way, e.g. checking `uct_tl_device_resource_t->type` for `UCT_DEVICE_TYPE_ACC`?

Reply:
Done